Patch by MP-Greg
Windows allows paths to use '/' for a path separator, adjust code to allow for
it
diff --git a/lib/rubygems/installer.rb b/lib/rubygems/installer.rb
index 8cb89e2e40..2e803be182 100644
--- a/lib/rubygems/installer.rb
+++ b/lib/rubygems/installer.rb
@@ -698,10 +698,12 @@ def process_options # :nodoc:
 
   def check_that_user_bin_dir_is_in_path # :nodoc:
     user_bin_dir = @bin_dir || Gem.bindir(gem_home)
-    user_bin_dir = user_bin_dir.gsub(File::SEPARATOR, File::ALT_SEPARATOR) if
-      File::ALT_SEPARATOR
-
     path = ENV['PATH']
+    if File::ALT_SEPARATOR
+      user_bin_dir = user_bin_dir.gsub(File::SEPARATOR, File::ALT_SEPARATOR)
+      path = path.gsub(File::SEPARATOR, File::ALT_SEPARATOR)
+    end
+    
     if Gem.win_platform? then
       path = path.downcase
       user_bin_dir = user_bin_dir.downcase
